specs/Standards/MultiToken/Core.md

```diff
@@ -0,0 +1,355 @@
+// function call. It allows you to attach any Multi Token, token in a call to a
+// receiver contract.
+//
+// Requirements:
+// * Caller of the method must attach a deposit of 1 yoctoⓃ for security
+//   purposes
+// * Caller must have greater than or equal to the `amount` being requested
+// * Contract MUST panic if called by someone other than token owner or,
+//   if using Approval Management, one of the approved accounts
+// * The receiving contract must implement `mt_on_transfer` according to the
+//   standard. If it does not, MT contract's `mt_resolve_transfer` MUST deal
+//   with the resulting failed cross-contract call and roll back the transfer.
+// * Contract MUST implement the behavior described in `mt_resolve_transfer`
+// * `approval_id` is for use with Approval Management extension, see
+//   that document for full explanation.
+// * If using Approval Management, contract MUST nullify approved accounts on
+//   successful transfer.
+// * Contract MUST panic if called with the length of `token_ids` not equal to `amounts` is not equal
+// * Contract MUST panic if `approval_ids` is not `null` and does not equal the length of `token_ids`
+//
+// Arguments:
+// * `receiver_id`: the valid NEAR account receiving the token.
+// * `token_ids`: the tokens to transfer
+// * `amounts`: the number of tokens to transfer, wrapped in quotes and treated
+//    like an array of string, although the numbers will be stored as an array of
+//    unsigned integer with 128 bits.  
+// * `approval_ids`: expected approval IDs per `token_ids`. If a `token_id` does 
+//    not have a corresponding approval id then the entry in the array must be marked null.
+//    The `approval_ids` are numbers smaller than 2^53, and therefore representable as JSON. 
+//    ApprovalId See Approval Management standard for full explanation.
+// * `memo` (optional): for use cases that may benefit from indexing or
+//    providing information for a transfer.
+// * `msg`: specifies information needed by the receiving contract in
+//    order to properly handle the transfer. Can indicate both a function to
+//    call and the parameters to pass to that function.
+
+
+function mt_batch_transfer_call(
+  receiver_id: string,
+  token_ids: string[],
+  amounts: string[],
+  approval_ids: (number|null)[] | null,
+  memo: string|null,
+  msg: string,
+): Promise {}
+
+/****************/
+/* VIEW METHODS */
+/****************/
+
+
+// Returns the tokens with the given `token_ids` or `null` if no such token.
+function mt_token(token_ids: string[]) (Token | null)[]
+
+// Returns the balance of an account for the given `token_id`.  
+// The balance though wrapped in quotes and treated like a string, 
+// the number will be stored as an unsigned integer with 128 bits.
+// Arguments:
+// * `account_id`: the NEAR account that owns the token.
+// * `token_id`: the token to retrieve the balance from
+function mt_balance_of(account_id: string, token_id: string): string
```

Comment:
does this just return `1` if the account has an NFT with the passed in token_id and 0 otherwise? 

For FTs, it would return the number of FTs that the account has for the given token_id correct?

Reply:
yes